httpclient5/src/main/java/org/apache/hc/client5/http/impl/auth/HttpAuthenticator.java

```diff
@@ -97,32 +101,17 @@ public boolean isChallenged(
             final HttpResponse response,
             final AuthExchange authExchange,
             final HttpContext context) {
-        final int challengeCode;
-        switch (challengeType) {
-            case TARGET:
-                challengeCode = HttpStatus.SC_UNAUTHORIZED;
-                break;
-            case PROXY:
-                challengeCode = HttpStatus.SC_PROXY_AUTHENTICATION_REQUIRED;
-                break;
-            default:
-                throw new IllegalStateException("Unexpected challenge type: " + challengeType);
-        }
-
-        final HttpClientContext clientContext = HttpClientContext.cast(context);
-        final String exchangeId = clientContext.getExchangeId();
-
-        if (response.getCode() == challengeCode) {
-            if (LOG.isDebugEnabled()) {
-                LOG.debug("{} Authentication required", exchangeId);
-            }
+        if (checkChallenged(challengeType, response, context)) {
             return true;
         }
         switch (authExchange.getState()) {
         case CHALLENGED:
         case HANDSHAKE:
             if (LOG.isDebugEnabled()) {
-                LOG.debug("{} Authentication succeeded", exchangeId);
+                final HttpClientContext clientContext = HttpClientContext.cast(context);
+                final String exchangeId = clientContext.getExchangeId();
+                // The mutual auth may still fail
+                LOG.debug("{} Server has accepted authorization", exchangeId);
```

Comment:
authentication?

Reply:
Technically, the client sends WWW-Authenticate headers, but the client sends Authorization headers.
At this point, the server has authenticated AND authorized the client, so I think that the message is correct.

But we can of course change the text.

Reply:
I know, but then it is inconsistent with other log messages.